### PR TITLE
[SDK-3909] visionOS initial documentation

### PR DESCRIPTION
**Pull request by lowip**
# Pull Request/Issue Resolution

#### Description of Change:

Adding a documentation page describing the supported, partially supported and unsupported features for visionOS

#### Is this change associated with a Braze feature/product release?
- [x] No

#### For reviewers

Not sure if we should add anything else, let me know if you have suggestions.

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [x] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [x] Check that all links work.
- [x] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @josh-mccrowell-braze and @internetisaiah as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] If the documentation involves a 1) paid SKU, 2) a third party, 3) SMS, 4) AI, or 5) privacy, ensure that Ana Teresa Serafino on the Legal team has signed off.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @josh-mccrowell-braze or @internetisaiah for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs<br>Currents</td>
  </tr>
  <tr>
    <td>@internetisaiah</td>
    <td>Developer Guide<br>Technology partners<br>Channels</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Intelligence<br>In-App Messages<br>Channels<br>Frontend Infrastructure & Experience (FIX)</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging Experience<br>Message Components<br>Email (Composition and Infrastructure) (tag @rachel-feinberg for Liquid use cases)</td>
  </tr>
  <tr>
    <td>@rachel-feinberg</td>
    <td>Customer Lifecycle, Identity and Permissions<br>SMS<br>User Targeting<br>Reporting</td>
  </tr>
</table>
</details>

### CODE REVIEW

**Comment by josh-mccrowell-braze** [comment]
Assigning @internetisaiah as the reviewer when the PR is ready to be moved out of draft.

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+article_title: visionOS
+page_order: 7.2
+platform: 
+  - iOS
+description: "visionOS support"
+---
+
+# visionOS
+
+With the release of the Apple Vision Pro, Apple has introduced a new platform called visionOS. Starting with the Braze Swift SDK [8.0.0](https://github.com/braze-inc/braze-swift-sdk/blob/main/CHANGELOG.md#800), visionOS is supported as a first class platform.
```
Comment (jacksonemiller) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:12]:
```suggestion
With the release of the Apple Vision Pro, Apple has introduced a new platform called visionOS. Starting with the Braze Swift SDK [8.0.0](https://github.com/braze-inc/braze-swift-sdk/blob/main/CHANGELOG.md#800), visionOS is supported as a first-class platform.
```

**Review by jacksonemiller — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- Push Stories
+  - The _Next_ button to navigate between Push Story pages is not working as intended
+  - Scrolling and clicking on the Push Story page works as expected
+
+We expect that future visionOS updates will address these limitations.
+
+## Unsupported Features
+
+Currently, Geofences Monitoring is not supported. The geofences related CoreLocation APIs are not made available by Apple on visionOS.
+
```
Comment (jerielng) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:42]: We could also add a note that Live Activities are not supported as well:
```suggestion
- Live Activities are not supported, as ActivityKit is only available on iOS and iPadOS.
```

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  - Audio playback is not supported
+- Push Stories
+  - The _Next_ button to navigate between Push Story pages is not working as intended
+  - Scrolling and clicking on the Push Story page works as expected
+
+We expect that future visionOS updates will address these limitations.
+
+## Unsupported Features
+
+Currently, Geofences Monitoring is not supported. The geofences related CoreLocation APIs are not made available by Apple on visionOS.
```
Comment (jerielng) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:41]:
```suggestion
- Currently, Geofences Monitoring is not supported. The Core Location APIs for region monitoring are not made available by Apple on visionOS.
```

**Review by jerielng — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- Push Notifications
+- Feature Flags
+- Location Analytics
+
+## Partially Supported Features
+
+The following features are partially supported on visionOS:
+- Rich Push Notifications
+  - Images display as expected
+  - GIFs and videos display the preview thumbnail, it is not currently possible to play the media
```
Comment (hokstuff) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:31]:
```suggestion
  - GIFs and videos display the preview thumbnail, but it is not currently possible to play the media
```

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+article_title: visionOS
+page_order: 7.2
+platform: 
+  - iOS
+description: "visionOS support"
+---
+
+# visionOS
+
+With the release of the Apple Vision Pro, Apple has introduced a new platform called visionOS. Starting with the Braze Swift SDK [8.0.0](https://github.com/braze-inc/braze-swift-sdk/blob/main/CHANGELOG.md#800), visionOS is supported as a first-class platform.
```
Comment (hokstuff) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:12]:
```suggestion
With the release of the Apple Vision Pro, Apple has introduced a new platform called [visionOS](https://developer.apple.com/visionos/). Starting with the Braze Swift SDK [8.0.0](https://github.com/braze-inc/braze-swift-sdk/blob/main/CHANGELOG.md#800), visionOS is supported as a first-class platform.
```

**Review by hokstuff — APPROVED** [review]
Minor nits - LGTM!

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+> Please refer to our [sample apps]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/sample_apps/) for working examples of visionOS integration.
+
+## Supported Features
+
+Most of the features available on iOS are supported on visionOS. This includes:
+- Analytics (sessions, custom events, purchases, etc.)
+- In-App Messaging (data models and UI)
+- Content Cards (data models and UI)
+- Push Notifications
```
Comment (jocelynzz) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:22]: Do we need to clarify this refers to "regular push"?

**Review by jocelynzz — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+---
+nav_title: visionOS
+article_title: visionOS
+page_order: 7.2
+platform: 
+  - iOS
+description: "visionOS support"
```
Comment (vanessaland) [inline _docs/_developer_guide/platform_integration_guides/swift/visionos.md:7]: Maybe we can put a sentence like this to describe the page?:
```suggestion
description: "This article covers the features supported on visionOS."
```

**Review by vanessaland — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/swift/visionos.md (hunk 6 above)
Comment (lowip) [reply _docs/_developer_guide/platform_integration_guides/swift/visionos.md:22]:
```suggestion
- Push Notifications (user-visible, with action buttons and silent notifications)
```

**Comment by lowip** [comment]
Hey @internetisaiah, this is ready for your review. You can merge it as your convenience.

**Comment by internetisaiah** [comment]
Awesome! I'll start reviewing today!

**Comment by internetisaiah** [comment]
@cla-bot check

**Comment by internetisaiah** [comment]
@cla-bot check

**Pull request merged**